lib/formz.dart

```diff
@@ -118,66 +74,58 @@ abstract class FormzInput<T, E> {
   /// For subsequent changes (in response to user input), the
   /// `FormzInput.dirty` constructor should be used to signify that
   /// the `FormzInput` has been manipulated.
-  final bool pure;
-
-  /// The [FormzInputStatus] which can be one of the following:
-  /// * [FormzInputStatus.pure]
-  ///   - if the input has not been modified.
-  /// * [FormzInputStatus.invalid]
-  ///   - if the input has been modified and validation failed.
-  /// * [FormzInputStatus.valid]
-  ///   - if the input has been modified and validation succeeded.
-  FormzInputStatus get status => pure
-      ? FormzInputStatus.pure
-      : valid
-          ? FormzInputStatus.valid
-          : FormzInputStatus.invalid;
-
-  /// Returns a validation error if the [FormzInput] is invalid.
-  /// Returns `null` if the [FormzInput] is valid.
-  E? get error => validator(value);
+  final bool isPure;
 
   /// Whether the [FormzInput] value is valid according to the
   /// overridden `validator`.
   ///
   /// Returns `true` if `validator` returns `null` for the
   /// current [FormzInput] value and `false` otherwise.
-  bool get valid => validator(value) == null;
+  bool get isValid => validator(value) == null;
 
   /// Whether the [FormzInput] value is not valid.
   /// A value is invalid when the overridden `validator`
   /// returns an error (non-null value).
-  bool get invalid => status == FormzInputStatus.invalid;
+  bool get isNotValid => !isValid;
+
+  /// Returns a validation error if the [FormzInput] is invalid.
+  /// Returns `null` if the [FormzInput] is valid.
+  E? get error => validator(value);
 
   /// A function that must return a validation error if the provided
   /// [value] is invalid and `null` otherwise.
   E? validator(T value);
 
+  /// The error to display if the [FormzInput] value
+  /// is not valid and has been modified.
+  E? get displayError => isPure ? null : error;
```

Comment:
Naming things is hard! I can see this as a temptation to use it directly in UI while bypassing `l10n`.

Not a big deal, but something to consider.

Reply:
Yeah completely agree and open to other suggestions 👍

Reply:
Hmm, maybe a boolean `shouldDisplayError`?

Reply:
I propose leaving it as is for now and we can revisit if needed